src/components/post-card.tsx

```diff
@@ -1,24 +1,81 @@
+'use client';
+
 import { Card, CardContent, CardHeader } from './ui/card';
-import { Avatar, AvatarFallback, AvatarImage } from './ui/avatar';
 import { Button } from './ui/button';
-import { MessageCircle, Heart, Repeat2 } from 'lucide-react';
+import { Drawer, DrawerContent, DrawerTitle, DrawerTrigger } from './ui/drawer';
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from '@/components/ui/popover';
+import { getTimeDistance } from '../lib/utils';
+import { UserIcon } from './user-icon';
+import {
+  Bookmark,
+  EyeOff,
+  Heart,
+  Link2,
+  MessageCircle,
+  MessageSquareOff,
+  MoreHorizontal,
+  Repeat2,
+  Share,
+  ShieldAlert,
+  UserRoundX,
+} from 'lucide-react';
 import Link from 'next/link';
+import { useMediaQuery } from '@mui/material';
 import type { Post } from '@/types/post';
+import type { UserIconInfo } from '@/types/user-icon-info';
 
 export const PostCard = ({ post }: { post: Post }) => {
+  const isDesktop = useMediaQuery('(min-width: 768px)');
+  const userIconInfo: UserIconInfo = {
+    user_id: post.user_id,
+    image: post.user.image,
+    is_following: post.user.is_following,
+  };
+
   return (
-    <Card className='w-2xl rounded-none p-4'>
-      <div className='gap-4'>
+    <Card className='w-full rounded-none p-4 pb-2 sm:w-2xs md:w-2xl'>
+      <div className='flex gap-4'>
+        <UserIcon iconInfo={userIconInfo} />
```

Comment:
userIconInfoという変数を宣言するのではなくスプレッド構文で書くほうが簡潔。
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Spread_syntax
今回であれば下のような実装がいい。
```suggestion
        <UserIcon iconInfo={...post.user} />
```